includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,313 @@
+	 *
+	 * @param bool|string $include        Comma separated IDs of post if fixing post content for multiple posts.
+	 * @param bool        $all            Flag for running the fix on all posts.
+	 * @param string      $post_type      The post type.
+	 * @param int         $posts_per_page The batch size.
+	 * @param int         $post_offset    The post offset.
+	 * @param bool        $dry_run        Flag for dry or normal run.
+	 * @param array       $args           WP CLI arguments.
+	 *
+	 * @return array|void
+	 */
+	public function fix( $include, $all, $post_type, $posts_per_page, $post_offset, $dry_run, $args = array() ) {
+		$this->dry_run = $dry_run;
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::log( __( 'Checking post content...', 'insecure-content-warning' ) );
+		}
+
+		if ( false === $all && ! empty( $args[0] ) ) {
+			$this->fix_insecure_content( $args[0] );
+		} elseif ( false === $all && empty( $args ) && false !== $include ) {
+			$post_ids_list = explode( ',', $include );
+			foreach ( $post_ids_list as $icw_post_id ) {
+				$this->fix_insecure_content( trim( $icw_post_id ) );
+			}
+			$this->total_post_count = count( $post_ids_list );
+		} else {
+			$total = 0;
+
+			// Loop through all posts and fix content.
+			while ( true ) {
+				$args  = array(
+					'posts_per_page' => $posts_per_page,
+					'post_type'      => $post_type,
+					'offset'         => $post_offset,
+					'orderby'        => 'ID',
+					'order'          => 'ASC',
+				);
+				$query = new WP_Query( $args );
+
+				if ( $query->have_posts() ) {
+					while ( $query->have_posts() ) {
+						$query->the_post();
+						$this->fix_insecure_content( get_the_ID() );
+					}
+				} else {
+					break;
+				}
+
+				// Set offset for next page.
+				$post_offset += $posts_per_page;
+				$total       += $query->post_count;
+
+				if ( ! defined( 'WP_CLI' ) ) {
+					// when fixing post in a REST request, do it only for the current batch and not until all the posts
+					// have been fixed.
+					break;
+				}
+
+				// Wait for a while before fixing the rest.
+				usleep( 1000 );
+			}
+
+			$this->total_post_count = $total;
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			// translators: Message to show when the fixing of insecure content is completed.
+			$message = PHP_EOL . sprintf( __( 'Total posts checked for insecure URL(s): %s', 'insecure-content-warning' ), $this->total_post_count ) . PHP_EOL;
+			WP_CLI::log( WP_CLI::colorize( "%c{$message}%n " ) );
+			Utils\format_items( 'table', $this->fixed_post_count, array( 'URL(s) fixed summary' ) );
+		} else {
+			if ( null === $this->fixed_post_count ) {
+				return __( 'No post(s) found', 'insecure-content-warning' );
+			}
+
+			$output = '';
+
+			foreach ( $this->warning_count as $warning ) {
+				$warning_message = array_shift( $warning );
+				$output         .= '<span class="warning">' . __( 'Warning', 'insecure-content-warning' ) . '</span>: ' . $warning_message . PHP_EOL;
```

Comment:
```suggestion
				$output         .= '<span class="warning">' . __( 'Warning', 'insecure-content-warning' ) . '</span>: ' . wp_strip_all_tags( $warning_message ) . PHP_EOL;
```

I know it isn't needed now but it protects ourselves from our future selves.

Reply:
Better safe than sorry, thank you.